pylint/checkers/utils.py

```diff
@@ -849,9 +849,12 @@ def uninferable_final_decorators(
     for decorator in getattr(node, "nodes", []):
         if isinstance(decorator, nodes.Attribute):
             try:
-                import_node = decorator.expr.lookup(decorator.expr.name)[1][0]
+                _, import_nodes = decorator.expr.lookup(decorator.expr.name)
             except AttributeError:
```

Comment:
Looking at the failed test this should be fixable with an `isinstance` on `decorator.expr`. I think we assume `expr` to a `LocalsDictNodeNG` while apparently in those regression tests it is a `nodes.Attribute`. Should we try and fix that while we're at it?
In general I prefer explicitness over broader `try...excepts`.

Reply:
`decorator.expr` has no attribute `loookup`. I couldn't try adding a `hasattr` instead of an `isinstance` if I understood you.

Reply:
But normally it does right? My hunch is that this code works we for `@decorator.func()` but not for `@decorator.module.func()`. Apparently sometimes `expr` sometimes does have `lookup` so instead of using `hasattr` can't we reliable determine what instance expr has and whether that instance is supposed to have lookup?